Content.Shared/StationRecords/GeneralStationRecordsFilter.cs

```diff
@@ -0,0 +1,83 @@
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.StationRecords;
+
+[Serializable, NetSerializable]
+public sealed class GeneralStationRecordsFilter
+{
+    public GeneralStationRecordFilterType type { get; set; }
+        = GeneralStationRecordFilterType.Name;
+    public string value { get; set; } = "";
+    public GeneralStationRecordsFilter(GeneralStationRecordFilterType filterType, string newValue = "")
+    {
+        type = filterType;
+        value = newValue;
+    }
+
+    public bool IsSkippedRecord(GeneralStationRecord someRecord)
```

Comment:
no - what i mean by this is that you should have separate objects for each of these cases, and each of those objects has a method that returns a boolean for the input value passed into it

imagine that this is expanded further to include more than just these three categories: what would it look like if we had 10 fields to search through?

Reply:
for 10 fields it looks well. The enum will be added new fields and the method "IsSkippedRecord" will be expanded with new fields. Everything will be changed in the file "GeneralStationRecordsFilter". Sо, I think, it is looks good. The other case, it can't work with an union of fields in filter records. But current UI is this kind.

Reply:
ok, so just move this back to the server-side record console system and shrink that switch case down into a value switch: i.e.,

```cs
return type switch
{
  GeneralStationRecordFilterType.Name => // filter out name field
  ...
}
```

that way you can just directly return if something can be filtered out or not instead of having to have redundant variables for it

Reply:
i did it